src/iqm_client/iqm_client.py

```diff
@@ -550,27 +574,39 @@ def submit_circuits(
         Returns:
             ID for the created task. This ID is needed to query the status and the execution results.
         """
-        serialized_qubit_mapping: Optional[list[SingleQubitMapping]] = None
-        if qubit_mapping is not None:
-            # check if qubit mapping is injective
-            target_qubits = set(qubit_mapping.values())
-            if not len(target_qubits) == len(qubit_mapping):
-                raise ValueError('Multiple logical qubits map to the same physical qubit.')
-
-            # check if qubit mapping covers all qubits in the circuits
-            for circuit in circuits:
+        serialized_qubit_mappings: Optional[BatchQubitMapping] = None
+        if qubit_mappings is not None:
+            # check that the number of qubit mappings matches the number of circuits
+            if len(qubit_mappings) != len(circuits):
+                raise ValueError('The number of qubit mappings does not match the number of circuits.')
+
+            # check if all qubit mappings map to same physical qubits
+            physical_qubits = set(next(iter(qubit_mappings)).values())
+            if not all(set(qubit_mapping.values()) == physical_qubits for qubit_mapping in qubit_mappings):
```

Comment:
```suggestion
            if len(set(frozenset(m.values()) for m in qubit_mappings)) != 1:
```
Maybe more pythonic?

Reply:
Right, I totally forgot about `frozenset`. I tried this with set and obviously it didn't work since set is not hashable